rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/InitScriptGenerator.java

```diff
@@ -0,0 +1,133 @@
+ */
+package org.ow2.proactive.resourcemanager.utils;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.log4j.Logger;
+
+import com.google.common.base.Joiner;
+
+
+public class InitScriptGenerator {
+
+    public static final String DETACHED_MODE_PREFIX_PROPERTY = "%detachedModePrefix%";
+
+    public static final String JAVA_PATH_PROPERTY = "%javaPath%";
+
+    public static final String SCHEDULING_PATH_PROPERTY = "%schedulingPath%";
+
+    public static final String FILE_ENCODING = "%fileEncoding%";
+
+    public static final String JAVA_OPTIONS_PROPERTY = "%javaOptions%";
+
+    public static final String RM_URL_PROPERTY = "%rmUrl%";
+
+    public static final String NODE_NAME_PROPERTY = "%nodeName%";
+
+    public static final String NODE_SOURCE_NAME_PROPERTY = "%nodeSourceName%";
+
+    public static final String CREDENTIALS_PROPERTY = "%credentials%";
+
+    public static final String NUMBER_OF_NODES_PER_INSTANCE_PROPERTY = "%numberOfNodesPerInstance%";
+
+    private static final Logger logger = Logger.getLogger(InitScriptGenerator.class);
+
+    protected static Configuration nsConfig;
+
+    static {
+        try {
+            //load configuration manager with the NodeSource properties file
+            nsConfig = NodeCommandLineProperties.loadConfig();
+        } catch (ConfigurationException e) {
+            logger.error("Exception when loading NodeSource properties", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static String defaultLinuxAgentCommand = nsConfig.getString(NodeCommandLineProperties.LINUX_AGENT_STARTUP_COMMAND);
+
+    /*
+     * TODO:
+     * Once the command template is ready in the properties file replace LINUX_AGENT_STARTUP_COMMAND
+     * with LINUX_JAR_STARTUP_COMMAND, to be done in the next step
+     */
+    private static String defaultLinuxJarCommand = nsConfig.getString(NodeCommandLineProperties.LINUX_AGENT_STARTUP_COMMAND);
+
+    /**
+     * Fills the properties of the default Linux startup script with variables collected by {@link CommandLineBuilder}.
+     * @param deploymentMode the mode of deployment ("Agent" or "Jar").
+     * @param startupScript the Linux startup script added by the user in the GUI.
+     * @param javaPath the path to the bin java.
+     * @param schedulingPath the path to the rm home.
+     * @param rmUrl the Url of the RM (pamr://0/ by default).
+     * @param fileEncoding the file encoding (UTF-8 by default).
+     * @param javaOptions the options to be passed to the Java command.
+     * @param nodeName the name of the node.
+     * @param nodeSourceName the name of the node source.
+     * @param credentials the RM credentials or an obfuscated credentials String .
+     * @param detached if set to true, the node agent will ignore the RM termination signal.
```

Comment:
This comment is a bit confusing for me. Does `nohup` make agent ignore RM termination signal? Will it impact remove the node source? And what about nodes started through jar?